storage/src/robocorp/storage/_client.py

```diff
@@ -0,0 +1,170 @@
+
+LOGGER = logging.getLogger(__name__)
+
+# Assets under `DATA_LIMIT` can be uploaded directly through the POST request,
+# instead of requesting a separate upload url
+DATA_LIMIT = 5 * 10**6  # 5MB
+
+
+def _estimate_base64_size(size: int) -> int:
+    """Return guess for content length after being base64 encoded."""
+    return math.ceil(size / 3) * 4
+
+
+class AssetNotFound(HTTPError):
+    """No asset with given name/id found."""
+
+
+class AssetUploadFailed(RuntimeError):
+    """There was an unexpected error while uploading an asset."""
+
+
+class AssetsClient:
+    def __init__(self, workspace: str, endpoint: str, token: str):
+        self._endpoint = endpoint
+        self._workspace = workspace
+        self._token = token
+
+        route_prefix = url_join(self._endpoint, "workspaces", self._workspace, "assets")
+        default_headers = {
+            "Authorization": f"Bearer {self._token}",
+            "Content-Type": "application/json",
+        }
+
+        self._client = Requests(route_prefix, default_headers=default_headers)
+
+    def _handle_asset_not_found(self, asset_id: str):
+        """Create an error handler that raises `AssetNotFound` for 404 status codes."""
+
+        def _handle_error(response: Response):
+            try:
+                self._client.handle_error(response)
+            except HTTPError as exc:
+                if exc.status_code == 404:
+                    # NOTE(cmin764): Any `404 Not Found` error won't be retried
+                    #  and will bubble-up from the first request.
+                    raise AssetNotFound(
+                        f"Asset with id {asset_id!r} not found",
+                        status_code=exc.status_code,
+                        reason=exc.reason,
+                    ) from exc
+                raise
+
+        return _handle_error
+
+    def list_assets(self) -> list[Asset]:
+        """Retrieve list of assets."""
+        return self._client.get("").json()
+
+    def create_asset(self, name: str) -> AssetDetails:
+        """Create a new asset."""
+        payload = {"name": name}
+        response = self._client.post("", json=payload)
+        return cast(AssetDetails, response.json())
+
+    def get_asset(self, asset_id: str) -> AssetDetails:
+        """Returns the details of a single asset, including its payload."""
+        path = quote_plus(asset_id)
+        handler = self._handle_asset_not_found(asset_id)
+        response = self._client.get(path, _handle_error=handler)
+        return cast(AssetDetails, response.json())
+
+    def delete_asset(self, asset_id: str):
+        """Delete the given asset."""
+        path = quote_plus(asset_id)
+        handler = self._handle_asset_not_found(asset_id)
+        self._client.delete(path, _handle_error=handler)
+
+    def upload_asset(
+        self,
+        asset_id: str,
+        content: bytes,
+        content_type: Optional[ContentType] = None,
+        wait=True,
+        poll_interval=0.5,
```

Comment:
```suggestion
        wait: bool = True,
        poll_interval: float = 0.5,
```

Reply:
mypy (and other tools) do correct type inference from defaults, but again, I don't have a strong preference here if you think being explicit is better

Reply:
That's interesting, coming from RF where in many cases being explicit was better than implicit and my first reaction was that you forgot to put the types in :).

Might be better to be explicit, which will help with those cases when you come back to the code to edit a parameter and you edit the default into a `None` (or something that isn't trivial for a human to guess, like a value returned by a function call) and you forget to update the type then.
Also being explicit might help with docs generation or IDE auto-completion in those scenarios where the tool is not as smart as **mypy**.